app-misc/kmonad/kmonad-0.4.1.ebuild

```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# ebuild generated by hackport 0.7.2.2
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour"
+inherit haskell-cabal
+
+DESCRIPTION="Advanced keyboard remapping utility"
+HOMEPAGE="https://hackage.haskell.org/package/kmonad"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=dev-haskell/cereal-0.5.8.1:=[profile?] <dev-haskell/cereal-0.6:=[profile?]
+		>=dev-haskell/lens-4.17.1:=[profile?] <dev-haskell/lens-6:=[profile?]
+		>=dev-haskell/megaparsec-7.0.5:=[profile?] <dev-haskell/megaparsec-10:=[profile?]
+		>=dev-haskell/optparse-applicative-0.14.3.0:=[profile?] <dev-haskell/optparse-applicative-0.17:=[profile?]
+		>=dev-haskell/resourcet-1.2.2:=[profile?] <dev-haskell/resourcet-1.3:=[profile?]
+		>=dev-haskell/rio-0.1.14.0:=[profile?] <dev-haskell/rio-0.2:=[profile?]
+		>=dev-haskell/unliftio-0.2.12:=[profile?] <dev-haskell/unliftio-0.3:=[profile?]
+		>=dev-lang/ghc-8.6.3:=[profile?] <dev-lang/ghc-9:=[profile?]
```

Comment:
Unfortunately, this won't have the effect that you want it to. Instead of refusing to install `kmonad` when the user has `>=ghc-9` installed, it will try to downgrade to an earlier `ghc` when they go to install `kmonad`.

You can see how this would be a problem.

If the latest release on hackage won't build on our current GHC, we can use a working snapshot from the git repo instead. Take a look at the latest [`stack` ebuild](https://github.com/gentoo-haskell/gentoo-haskell/blob/master/dev-haskell/stack/stack-2.7.6_pre20220812.ebuild) for an example of how to do that.

Reply:
You should be able to copy your live ebuild and modify it so it uses a snapshot from github instead of `git-r3.eclass`, and `KEYWORDS="~amd64 ~x86"`. Just make sure to name it something like `kmonad-<VER>-pre_<DATE>.ebuild`, where

- `<VER>` is the version listed in the cabal file from the snapshot. If it is still `0.4.1`, you can use `0.4.1.0.1` instead.
- `<DATE>` is the date of the snapshot you are using (for instance `20220826`)

Reply:
Thank you for your guidance. I was able to create that ebuild correctly with a snapshot from github.

Reply:
> `<VER>` is the version listed in the cabal file from the snapshot. If it is still 0.4.1, you can use 0.4.1.0.1 instead.

I suppose `0.4.1.1` would also work...

Anyway, it's not important. This looks good! I will merge it once I make sure it builds on my end.

Reply:
Let's go with shorter filename then.